datajoint/schema.py

```diff
@@ -1,15 +1,59 @@
+from operator import itemgetter
 import pymysql
 import logging
 
-from . import conn, DataJointError, NoDefinitionError
+from . import conn, DataJointError
+from datajoint.utils import to_camel_case
 from .heading import Heading
 from .base_relation import BaseRelation
-from .user_relations import Part
+from .user_relations import Part, Computed, Imported, Manual, Lookup
 import inspect
+
 logger = logging.getLogger(__name__)
 from warnings import warn
 
 
+class SchemaFactory:
```

Comment:
SchemaFactory might not be the most descriptive name.  This is effectively a function.  The only reason it's a class is because it has a state.  So we can use an action word.  I think I would call it something like `create_relation_class_from_database`.